app/models/operating_system_flavor.rb

```diff
@@ -1,6 +1,6 @@
 class OperatingSystemFlavor < ApplicationRecord
   acts_as_miq_taggable
-  belongs_to :provisioning_manager
+  belongs_to :provisioning_manager, :class_name => "ManageIQ::Providers::ProvisioningManager"
```

Comment:
I believe this is a problem because this class is outside of the ManageIQ::Provider namespace so it doesn't resolve the name solely by the name of the association.